miner/worker.go

```diff
@@ -534,12 +538,14 @@ func (w *worker) newWorkLoop(recommit time.Duration) {
 			w.clearPending(w.chain.CurrentBlock().Number.Uint64())
 
 			timestamp = time.Now().Unix()
+			w.pendingWorkBlock.Store(w.chain.CurrentBlock().Number.Uint64() + 1)
 			commit(false, commitInterruptNewHead)
 
 		case head := <-w.chainHeadCh:
 			w.clearPending(head.Header.Number.Uint64())
 
 			timestamp = time.Now().Unix()
+			w.pendingWorkBlock.Store(head.Header.Number.Uint64() + 1)
```

Comment:
We might need to check `pendingWorkBlock` here as well? It is possible that veblop timer creates a work task before entering here.